experimental/packages/opentelemetry-sdk-metrics-base/src/MeterProvider.ts

```diff
@@ -17,132 +17,118 @@
 import * as api from '@opentelemetry/api';
 import * as metrics from '@opentelemetry/api-metrics';
 import { Resource } from '@opentelemetry/resources';
-import { Measurement } from './Measurement';
 import { Meter } from './Meter';
 import { MetricExporter } from './MetricExporter';
 import { MetricReader } from './MetricReader';
-import { View } from './View';
+import { MeterProviderSharedState } from './state/MeterProviderSharedState';
+import { InstrumentSelector } from './view/InstrumentSelector';
+import { MeterSelector } from './view/MeterSelector';
+import { View } from './view/View';
+
 
 // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#meterprovider
 
 export type MeterProviderOptions = {
-    resource?: Resource;
+  resource?: Resource;
 }
 
 export class MeterProvider {
-    private _resource: Resource;
-    private _shutdown = false;
-    private _metricReaders: MetricReader[] = [];
-    private _metricExporters: MetricExporter[] = [];
-    private _views: View[] = [];
-
-    constructor(options: MeterProviderOptions) {
-        this._resource = options.resource ?? Resource.empty();
+  private _sharedState: MeterProviderSharedState;
+  private _shutdown = false;
+  private _metricReaders: MetricReader[] = [];
+  private _metricExporters: MetricExporter[] = [];
+
+  constructor(options: MeterProviderOptions) {
+    this._sharedState = new MeterProviderSharedState(options.resource ?? Resource.empty());
+  }
+
+  /**
+   * **Unstable**
+   *
+   * This method is only here to prevent typescript from complaining and may be removed.
+   */
+  getResource() {
+    return this._sharedState.resource;
+  }
+
+  getMeter(name: string, version = '', options: metrics.MeterOptions = {}): metrics.Meter {
```

Comment:
Most of these are whitespace changes: changed 4 spaces indentation to 2 spaces indentation.